### ISSUE TITLE
Dropdown

### ISSUE THREAD

**Issue by joshwooding** [jpmorganchase/salt-ds#2764]
We need to re-visit Dropdown to add missing features and update its API so it matches Salt's API goals.

```[tasklist]
### Tasks
- [ ] #2359 
- [ ] #2460 
- [ ] #2363 
- [ ] #2361 
- [ ] #2357 
- [ ] Update Dropdown to use declarative children
```

**Comment by joshwooding** [comment]
Copied from #2607

20 Nov - snapshot release last week, waiting for MTK feedback, build needs reviewing

**Comment by origami-z** [comment]
22 Nov - Adding some examples to the WIP Combobox and Dropdown and currently writing tests

**Comment by joshwooding** [comment]
24 Nov - Added `type="button"` but haven't worked on Dropdown much focus has been on ComboBox

**Comment by joshwooding** [comment]
1 Dec - Still haven't been able to work on this

**Comment by joshwooding** [comment]
18 Dec - Call tomorrow with stakeholders, @bhoppers2008 to take a quick look today and re-sync tomorrow

**Comment by joshwooding** [comment]
09/01 - Working on finishing addressing the feedback, have been raising separate PRs to fix issues and had to update floating-ui to solve some issues.

**Comment by joshwooding** [comment]
Closed by #3131
